File: 1353-find-resultant-array-after-removing-anagrams/find-resultant-array-after-removing-anagrams.py
class Solution:
    def removeAnagrams(self, words: List[str]) -> List[str]:
        if not words: return []
        res = []
        res.append(words[0])
        prev = Counter(words[0])
        for i in range(1,len(words)):
            cur = Counter(words[i])
            if prev != cur:
                res.append(words[i])
                prev = cur
        return res

File: 1353-find-resultant-array-after-removing-anagrams/find-resultant-array-after-removing-anagrams.cpp
class Solution {
public:
    vector<string> removeAnagrams(vector<string>& words) {
        vector<string> res;
        if (words.empty()) return res;
        res.push_back(words[0]);
        string prev = words[0];
        for (int i=1; i<words.size(); i++){
            string cur = words[i];
            int arr[26] = {0};
            for (char c:prev) arr[c-'a']++;
            for (char c:cur) arr[c-'a']--;
            bool valid = true;
            for (int f: arr){
                if (f != 0) {
                    valid = false;
                    break;
                }
            }
            if (!valid){ 
                res.push_back(cur);
                prev = cur;
            }
        }
        return res;
    }
};